### PR TITLE
Add option to include severity in table output

### PR DESCRIPTION
**Pull request by giovanni-bozzano**
Add an optional `--include-severity` flag to include a severity column when using table or markdown output, when severity is present in the OSV schema.

### CODE REVIEW

**Review by G-Rath — CHANGES_REQUESTED** [review]
Can you please add tests

**Comment by giovanni-bozzano** [comment]
Sure, test added.

From an end user perspective, it would be useful to also have the numeric score computed in the same column, for a quicker overview. Let me know if this sounds good.

**Comment by another-rex** [comment]
Thanks for working on this @giovanni-bozzano! 

Unfortunately we want to be very careful with what we add to the table output as we want to keep the horizontal width down, and also keep the number of CLI flags limited. 

Converting `CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:L` into a single numeric score will help, but it also loses a decent amount of information. 

A more long term solution is to create a markdown/text based output format, which is more suitable for adding additional information like severity info, fixed versions...etc, We are currently developing this as part of #57, currently estimated to be completed this quarter. 

For now a workaround would be to parse the json output to get the severity value.

**Comment by another-rex** [comment]
Discussed this a bit more with the team, something similar to this could work showing only the computed scores in the table, and only showing the full string (or a subset of it) in the text output I was describing above so if the user wants more info they can get it. 

We can use https://github.com/goark/go-cvss to compute the score relatively easily, if you can replace the current full severity string with just the score it would be great!

File: internal/output/table.go [hunk 1]
```diff
@@ -15,10 +19,15 @@ import (
...
 
 // PrintTableResults prints the osv scan results into a human friendly table.
-func PrintTableResults(vulnResult *models.VulnerabilityResults, outputWriter io.Writer) {
+func PrintTableResults(vulnResult *models.VulnerabilityResults, outputWriter io.Writer, includeSeverity bool) {
 	outputTable := table.NewWriter()
 	outputTable.SetOutputMirror(outputWriter)
-	outputTable.AppendHeader(table.Row{"OSV URL (ID In Bold)", "Ecosystem", "Package", "Version", "Source"})
+	row := table.Row{"OSV URL (ID In Bold)"}
+	if includeSeverity {
+		row = append(row, "Severity")
```
Comment (another-rex) [inline internal/output/table.go:27]: Can you rename severity in the output to "CVSS".

**Review by another-rex — CHANGES_REQUESTED** [review]
Thanks for the change! Agreed with adding to the documentation to clarify that this is the base severity score. 

I think now that it's reasonably short, we can remove the `include-severity` flag and just always show the severity column. (Does mean a couple of the tests in main_test.go will need to be updated)

**Comment by oliverchang** [comment]
> Thanks for the change! Agreed with adding to the documentation to clarify that this is the base severity score.
> 
> I think now that it's reasonably short, we can remove the `include-severity` flag and just always show the severity column. (Does mean a couple of the tests in main_test.go will need to be updated)

CC @hayleycd to see where we can fit this in our current docs.

**Comment by giovanni-bozzano** [comment]
I removed the flag and renamed the column to "CVSS". Tests have been changed accordingly.

File: internal/output/table.go [hunk 2]
```diff
@@ -100,6 +104,36 @@ func tableBuilderInner(vulnResult *models.VulnerabilityResults, addStyling bool,
...
+					for _, vuln := range pkg.Vulnerabilities {
+						if vuln.ID == vulnID {
+							severities = vuln.Severity
+						}
+					}
+					for i, severity := range severities {
+						if i != 0 {
+							outputSeverities = append(outputSeverities, ", ")
+						}
+
```
Comment (another-rex) [inline internal/output/table.go:119]:
```suggestion
```
I don't think this is correct, it'll append a new line to just a comma when you do a join on line 135. I think you just need to add `,`  to line 135

File: internal/output/table.go [hunk 3]
```diff
@@ -100,6 +104,36 @@ func tableBuilderInner(vulnResult *models.VulnerabilityResults, addStyling bool,
...
+							numericSeverity, _ := v3_metric.NewBase().Decode(severity.Score)
+							outputSeverity = fmt.Sprintf("%v", numericSeverity.Score())
+						default:
+							outputSeverity = severity.Score
+						}
+
+						outputSeverities = append(outputSeverities, outputSeverity)
+					}
+				}
+				outputRow = append(outputRow, strings.Join(outputSeverities, "\n"))
```
Comment (another-rex) [inline internal/output/table.go:135]:
```suggestion
				outputRow = append(outputRow, strings.Join(outputSeverities, ",\n"))
```
See other comment

**Review by another-rex — COMMENTED** [review]
Thanks! Just check if the final comments are correct then we should be good to merge!

File: internal/output/table.go (hunk 2 above)
Comment (giovanni-bozzano) [reply internal/output/table.go]: Right, thanks for the suggestion! Changed it in the latest commit.

**Comment by hayleycd** [comment]
So what determines if the severity score is CVSS v2 or CVSS v3? (Clarifying so I can update the docs)

**Comment by another-rex** [comment]
Inside each OSV entry the severity looks like this:
```
  "severity": [
    {
      "type": "CVSS_V3",
      "score": "CVSS:3.1/AV:N/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H"
    }
  ],
```

The type field shows what version is used, currently the schema defines only two options, v3 and v2: https://ossf.github.io/osv-schema/#severity-field

As for why you would choose one over the other I think it's just newer advisories use cvss v3, and older ones use v2.

**Review by another-rex — APPROVED** [review]

**Comment by hayleycd** [comment]
> 

So there is nothing specific to this code, it will just pull the cvss (whether version 2 or 3) from the OSV entry?

**Comment by another-rex** [comment]
Yep

**Pull request merged**